packages/esbuild-plugins/src/build.ts

```diff
@@ -0,0 +1,143 @@
+import * as process from "process";
+import { readFileSync } from "fs";
+import * as path from "path";
+import * as esbuild from "esbuild";
+import type { BuildOptions, Format } from "esbuild";
+import { umdWrapper } from "esbuild-plugin-umd-wrapper";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import { rebuildLogger } from "./rebuild-logger.ts";
+import { sfxWasm } from "./sfx-wrapper.ts";
+
+const myYargs = yargs(hideBin(process.argv));
+myYargs
+    .usage("Usage: node esbuild.mjs [options]")
+    .demandCommand(0, 0)
+    .example("node esbuild.mjs --watch", "Bundle and watch for changes")
+    .option("mode", {
+        alias: "m",
+        describe: "Build mode",
+        choices: ["development", "production"],
+        default: "production"
+    })
+    .option("w", {
+        alias: "watch",
+        describe: "Watch for changes",
+        type: "boolean"
+    })
+    .help("h")
+    .alias("h", "help")
+    .epilog("https://github.com/hpcc-systems/hpcc-js-wasm")
+    ;
+
+export const pkg = JSON.parse(readFileSync(path.join(process.cwd(), "./package.json"), "utf8"));
+export const NODE_MJS = pkg.type === "module" ? "js" : "mjs";
```

Comment:
Are these backwards, if type === module then mjs?

Reply:
No - if the type is "module" then it expects all "js" files to be modern "import" style.